store/cma_store.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+// cmaStore maintains the cumulative moving average of values at a specific Timestamp.
+// cmaStore is an implementation of TimeStore where the values are casted as uint64.
+type cmaStore struct {
+	buffer *list.List
+	rwLock sync.RWMutex
+}
+
+type TPContainer struct {
+	TimePoint
+	count uint16
```

Comment:
I'd say to just use an int, it is generally simpler and we don't need the optimization